include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
+[[vk::ext_instruction(spv::OpStore)]]
+enable_if_t<is_spirv_type_v<P>, void> store(P pointer, T object, [[vk::ext_literal]] uint32_t memoryAccess, [[vk::ext_literal]] uint32_t memoryAccessParam);
+
+template<typename T, typename P, uint32_t alignment>
+[[vk::ext_instruction(spv::OpStore)]]
+enable_if_t<is_spirv_type_v<P>, void> store(P pointer, T object, [[vk::ext_literal]] uint32_t __aligned = /*Aligned*/0x00000002, [[vk::ext_literal]] uint32_t __alignment = alignment);
+
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpStore)]]
+enable_if_t<is_spirv_type_v<P>, void> store(P pointer, T object);
+
+template<typename T>
+[[vk::ext_instruction(spv::OpStore)]]
+void store(pointer_t<spv::StorageClassPhysicalStorageBuffer, T> pointer, T object, [[vk::ext_literal]] uint32_t memoryAccess);
+
+template<typename T>
+[[vk::ext_instruction(spv::OpStore)]]
+void store(pointer_t<spv::StorageClassPhysicalStorageBuffer, T> pointer, T object, [[vk::ext_literal]] uint32_t memoryAccess, [[vk::ext_literal]] uint32_t memoryAccessParam);
+
+template<typename T, uint32_t alignment>
+[[vk::ext_instruction(spv::OpStore)]]
+void store(pointer_t<spv::StorageClassPhysicalStorageBuffer, T> pointer, T object, [[vk::ext_literal]] uint32_t __aligned = /*Aligned*/0x00000002, [[vk::ext_literal]] uint32_t __alignment = alignment);
+
+template<typename T>
+[[vk::ext_instruction(spv::OpStore)]]
+void store(pointer_t<spv::StorageClassPhysicalStorageBuffer, T> pointer, T object);
+
+template<typename T, typename P>
+[[vk::ext_instruction(spv::OpGenericPtrMemSemantics)]]
+enable_if_t<is_spirv_type_v<P>, T> genericPtrMemSemantics(P pointer);
+
+template<typename T>
+[[vk::ext_instruction(spv::OpGenericPtrMemSemantics)]]
+T genericPtrMemSemantics(pointer_t<spv::StorageClassPhysicalStorageBuffer, T> pointer);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitFieldInsert)]]
+T bitFieldInsert(T base, T insert, uint32_t offset, uint32_t count);
+
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitFieldSExtract)]]
+int32_t bitFieldExtract(int32_t base, uint32_t offset, uint32_t count);
+
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitFieldSExtract)]]
+int64_t bitFieldExtract(int64_t base, uint32_t offset, uint32_t count);
+
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitFieldUExtract)]]
+uint32_t bitFieldExtract(uint32_t base, uint32_t offset, uint32_t count);
+
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitFieldUExtract)]]
+uint64_t bitFieldExtract(uint64_t base, uint32_t offset, uint32_t count);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityBitInstructions)]]
+[[vk::ext_instruction(spv::OpBitReverse)]]
+T bitReverse(T base);
+
+template<typename T>
+[[vk::ext_instruction(spv::OpBitCount)]]
+T bitCount(T base);
```

Comment:
you probably want to express type constraints with `enable_if_t` on any templated instruction